packages/models-library/tests/test_service_settings_labels.py

```diff
@@ -0,0 +1,95 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+
+from pprint import pformat
+from typing import Any, Dict
+
+import pytest
+from models_library.service_settings_labels import (
+    SimcoreServiceLabels,
+    SimcoreServiceSettingLabelEntry,
+    SimcoreServiceSettingsLabel,
+)
+from pydantic import BaseModel
+
+
+def test_service_settings():
+    simcore_settings_settings_label = SimcoreServiceSettingsLabel.parse_obj(
+        SimcoreServiceSettingLabelEntry.Config.schema_extra["examples"]
+    )
+    assert simcore_settings_settings_label
+
+    # ensure private attribute assignment
+    for service_setting in simcore_settings_settings_label:
+        # pylint: disable=protected-access
+        service_setting._destination_container = "random_value"
+
+
+SIMCORE_SERVICE_EXAMPLES = [
+    (example, items, uses_dynamic_sidecar, index)
+    # pylint: disable=unnecessary-comprehension
+    for example, items, uses_dynamic_sidecar, index in zip(
+        SimcoreServiceLabels.Config.schema_extra["examples"],
+        [1, 2, 4],
+        [False, True, True],
+        ["legacy", "dynamic-service", "dynamic-service-with-compose-spec"],
+    )
+]
+
+
+@pytest.mark.parametrize(
+    "example, items, uses_dynamic_sidecar",
+    [
+        (example, items, uses_dynamic_sidecar)
+        for example, items, uses_dynamic_sidecar, _ in SIMCORE_SERVICE_EXAMPLES
+    ],
+    ids=[i for _, _, _, i in SIMCORE_SERVICE_EXAMPLES],
+)
+def test_simcore_service_labels(example: Dict, items: int, uses_dynamic_sidecar: bool):
+    simcore_service_labels = SimcoreServiceLabels.parse_obj(example)
+    assert simcore_service_labels
+    assert len(simcore_service_labels.dict(exclude_unset=True)) == items
+    assert simcore_service_labels.needs_dynamic_sidecar == uses_dynamic_sidecar
+
+
+@pytest.mark.parametrize(
+    "model_cls",
+    (
+        SimcoreServiceSettingLabelEntry,
+        SimcoreServiceSettingsLabel,
+        SimcoreServiceLabels,
+    ),
+)
+def test_service_settings_model_examples(
+    model_cls: BaseModel, model_cls_examples: Dict[str, Dict[str, Any]]
```

Comment:
Sorry, but that's the only one I was able to notice :( If there are more could you please tell me how you detect them? Because by looking at the code I'm not able to.